### PR TITLE
Extend PPA key submission, test result w/ RAT and PT(EXPOSUREAPP-6723)

### PR DESCRIPTION
**Pull request by chiljamgossow**
Part 1: Key submission
There are now two data sets / donors, one for PCR and one for RA tests. Both are extended with PT risk level metrics.

Part 2: Test result
same as above

Contains some bug fixes I realised when checking the code

TechSpec changes for PPA https://github.com/corona-warn-app/cwa-app-tech-spec/compare/release/2.1.x..proposal/ppa-rat#diff-efd86ae3917364d115b22667bfd7e2bbf780b1dfcc4ced613319e6e28a8f6af0

How to test: Register tests, submit in foreground/background, PPA data should contain 0-2 key submission metadata sets, at most one for each type. The types are differentiated with the flag submitted_after_rapid_antigen_test.

ToDo
- [ ] unit tests -> separate PR

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt [hunk 1]
```diff
@@ -15,98 +19,118 @@ import javax.inject.Inject
...
+                riskLevelSettings.ewMostRecentDateWithHighOrLowRiskLevel?.toLocalDateUtc(),
+                testRegisteredAt.toLocalDateUtc()
+            )
+        }
+
+        type.storage.ptDaysSinceMostRecentDateAtRiskLevelAtTestRegistration.update {
             calculateDaysSinceMostRecentDateAtRiskLevelAtTestRegistration(
-                riskLevelSettings.lastChangeCheckedRiskLevelTimestamp,
-                testRegisteredAt
+                riskLevelSettings.ptMostRecentDateWithHighOrLowRiskLevel,
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt:74]: Is this missing `toLocalDateUtc()`?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionStorage.kt [hunk 2]
```diff
@@ -6,78 +6,105 @@ import de.rki.coronawarnapp.util.preferences.clearAndNotify
...
 
-class AnalyticsKeySubmissionStorage @Inject constructor(
-    @AppContext val context: Context
+class AnalyticsPCRKeySubmissionStorage @Inject constructor(
+    @AppContext context: Context
+) : AnalyticsKeySubmissionStorage(context, "") // the original
+
+class AnalyticsRAKeySubmissionStorage @Inject constructor(
+    @AppContext context: Context
+) : AnalyticsKeySubmissionStorage(context, "_RAT")
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionStorage.kt:15]: Should we add a little test for this?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelSettings.kt [hunk 3]
```diff
@@ -49,13 +59,53 @@ class RiskLevelSettings @Inject constructor(
             putLong(PKEY_LAST_CHANGE_CHECKED_RISKLEVEL_TIMESTAMP_COMBINED, value?.millis ?: 0L)
         }
 
+    var ptMostRecentDateWithHighOrLowRiskLevel: LocalDate?
+        get() = prefs.getString(PKEY_MOST_RECENT_DATE_AT_RISK_LEVEL_PT, null).let {
+            return if (it != null) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelSettings.kt:64]: Something with `?.let` might have less nested null returns?

**Review by d4rken — COMMENTED** [review]
lgtm, did not test on device.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/rapidantigen/RapidAntigenProcessor.kt [hunk 4]
```diff
@@ -111,6 +121,10 @@ class RapidAntigenProcessor @Inject constructor(
                 }
             }
 
+            if (newTestResult == PCR_POSITIVE) {
+                analyticsKeySubmissionCollector.reportPositiveTestResultReceived(type)
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/rapidantigen/RapidAntigenProcessor.kt:125]: needs to be RAT

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionCollector.kt:82]: It is already a local date for PT. But I'm wondering if I even need this. I could just take it from RiskLevelStorage

**Review by kolyaopahle — DISMISSED** [review]
Logic looks good to me, was not able to test on device yet, will do tomorrow 👍

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionStorage.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/AnalyticsKeySubmissionStorage.kt:15]: Will add unit tests in a separate PR. I tested it manually

**Review by d4rken — APPROVED** [review]

**Comment by harambasicluka** [comment]
I merge it as @kolyaopahle isn't available. If he finds something this can be fixed in a followup PR.

**Pull request merged**